lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexer.java

```diff
@@ -22,6 +28,22 @@
 import io.pinecone.clients.Pinecone;
 import static io.pinecone.commons.IndexInterface.buildUpsertVectorWithUnsignedIndices;
 
+/**
+ *
+ * Config Parameters:
+ * - index (String) : index which PineconeIndexer will request from
+ * - namespaces (Map<String, Object>, Optional) : mapping of your namespace to the field of the document to retrieve the vectors from
+ * - apiKey (String) : API key used for requests
+ * - mode (String, Optional) : the type of request you want PineconeIndexer to send to your index
+ *    1. upsert (will replace if vector id exists in namespace or index if no namespace is given)
+ *    2. update (will only update embeddings)
+ *    3. delete (deletes all vectors given document ids)
+ *      - will only delete the exact id of documents that go through the pipeline, for id prefix deletion, look at deletionPrefix
+ * - defaultEmbeddingField (String, required if namespaces is not set) : the field to retrieve embeddings from
+ * - deleteByPrefix (boolean, optional) : sends a deletion by prefix request instead of a usual request
+ * - addDeletionPrefix (String, optional) : String that adds on to all document ids used for prefix deletion
+ *  - e.g. if id of doc is "doc1" & deletionPrefix is "-", then all vectors containing "doc1-" in id as prefix will be deleted.
```

Comment:
this is useful if there are multiple versions of documents in a namespace. e.g.  doc1-v1-chunk1 vs doc1-v2-chunk1. To delete all version 1 of doc1, here the user would put "-v1" in the addDeletionPrefix